x-pack/plugins/reporting/server/lib/tasks/error_logger.ts

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { LevelLogger } from '..';
+
+const MAX_PARTIAL_ERROR_LENGTH = 1000; // 1000 of beginning, 1000 of end
+const ERROR_PARTIAL_SEPARATOR = '...';
+const MAX_ERROR_LENGTH = MAX_PARTIAL_ERROR_LENGTH * 2 + ERROR_PARTIAL_SEPARATOR.length;
+
+/*
+ * An error message string could be very long, as it sometimes includes huge
+ * amount of base64
+ */
+export const errorLogger = (logger: LevelLogger, message: string, err?: Error) => {
```

Comment:
This function was copied nearly wholesale from ESQueue non-TS code, but the unit test is new.

It's not exactly a wholesale copy because the original code would include the stack trace in the limited-length error string. To make the function more testable, it only uses the "message" portion of the error object as the limited-length error string.